libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAbstractPasscodeViewController.m

```diff
@@ -57,28 +55,30 @@ @implementation SFAbstractPasscodeViewController
 
 @synthesize configData = _configData;
 @synthesize mode = _mode;
+@synthesize viewConfig = _viewConfig;
 
-- (id)initWithMode:(SFPasscodeControllerMode)mode passcodeConfig:(SFPasscodeConfigurationData)configData
+- (id)initWithMode:(SFPasscodeControllerMode)mode passcodeConfig:(SFPasscodeConfigurationData)configData {
+    return [self initWithMode:mode passcodeConfig:configData viewConfig:[SFSDKPasscodeViewConfig createDefaultConfig]];
+}
+            
+- (id)initWithMode:(SFPasscodeControllerMode)mode passcodeConfig:(SFPasscodeConfigurationData)configData viewConfig:(SFSDKPasscodeViewConfig *)config
 {
     self = [super init];
     if (self) {
         _mode = mode;
         _configData = configData;
+        _viewConfig = config;
         if (mode == SFPasscodeControllerModeCreate || mode == SFPasscodeControllerModeChange) {
-            NSAssert(_configData.passcodeLength > 0, @"You must specify a positive pin code length when creating a pin code.");
+            NSAssert(_configData.passcodeLength >= 0, @"You must specify a positive pin code length when creating a pin code.");
```

Comment:
Passcode length will be 0 when the actual passcode length is unknown.  This only happens as a one time upgrade step when transitioning from the old UI where we effectively stored minimum length (because it did not matter then) to the new UI where actual passcode length is necessary.